BLSQ/iaso
# IA-1601 query builder dates

Map output of query builder date to XLSFORM type and api format.

![Screenshot 2022-10-20 at 14 54 20](https://user-images.githubusercontent.com/12494624/197229960-fc2cd96a-ebb1-44cc-83f9-7f16a8bf48fc.png)

## Self proof reading checklist


## Changes

- moved instances utils to a folder
- parse jsonLogic value to change date value according to the type (date, dateTime, time)
- adapt displayed value using moment locale
- only use equal and not equal operators for those fields.


## How to test

Go to submission page.
Select a form with date field value
Open a query builder search and select a date.

date format should be 'YYYY-MM-DD'
dateTime => 'YYYY-MM-DD HH:mm'
Time => 'HH:mm'

You can verify the result in the payload or in the url

## Print screen / video

<img width="563" alt="Screenshot 2022-10-21 at 17 15 22" src="https://user-images.githubusercontent.com/12494624/197230150-393a04dc-0742-4320-b1f8-cbecf4c207f5.png">
<img width="1680" alt="Screenshot 2022-10-21 at 17 15 16" src="https://user-images.githubusercontent.com/12494624/197230154-9054661d-20ca-42a7-bbe8-c703b9ca14f6.png">

## Notes

I cannot fix the typing issue in `hat/assets/js/apps/Iaso/domains/instances/utils/jsonLogicParse.ts` line 44
